ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -478,17 +478,30 @@ private class PartitionStateChangeListenerImpl implements PartitionStateChangeLi
     // However, we need to make sure the internal state of store is only updated by primary cluster participant since
     // Replication Manager which only listens to primary participant also updates the internal store states. For such
     // needs, use this boolean.
-    private final boolean isPrimaryClusterParticipantListener;
+
+    /**
+     * Indicates whether it's a listener object for primary helix cluster-manager. Used to respond to state transition
+     * messages from primary helix cluster, which is simply the first string in zkConnectStr separated by commas.
+     * Renamed from isPrimaryClusterParticipantListener to isPrimaryClusterManagerListener because
+     * the usage of "participant" is completely wrong in the code at this point. It is being used to refer to both
+     * ambry-server node and helix. We don't want to fix all usages retroactively at this point,
+     * but going forward please follow this convention below.
+     * <p>
+     * participant = ambry-server node that _participates_ in cluster management done by helix cluster-manager service
+     * participant = _NOT_ helix cluster-manager service
+     * <p>
+     * And PartitionStateChangeListener listens to state transition messages from helix cluster-manager service.
+     */
```

Comment:
we don't need this explanation. I think this is confusing.

Reply:
fixed